packages/mobile/src/store/offline-downloads/sagas/utils/shouldCancelJob.ts

```diff
@@ -0,0 +1,38 @@
+import { reachabilityActions, reachabilitySelectors } from '@audius/common'
+import { NetInfoStateType } from '@react-native-community/netinfo'
+import { select, take } from 'typed-redux-saga'
+
+import {
+  getCurrentNetworkType,
+  getDownloadNetworkTypePreference
+} from '../../selectors'
+import {
+  setCurrentNetworkType,
+  setDownloadNetworkPreference
+} from '../../slice'
+const { SET_UNREACHABLE } = reachabilityActions
+const { getIsReachable } = reachabilitySelectors
+
+export function* shouldCancelJob() {
```

Comment:
love this omg!

Reply:
was this not also you? lol
I based all this off your branch `dj-cellular-download-option`

Reply:
wait really? omg it was so long ago